service/mapreduce_service.hh

```diff
@@ -141,7 +142,10 @@ public:
         , _proxy(p)
         , _db(db)
         , _shared_token_metadata(stm)
-        , _early_abort_subscription(as.subscribe([this] () noexcept { _shutdown = true; }))
+        , _early_abort_subscription(as.subscribe([this] () noexcept {
+            _shutdown = true;
+            _as.request_abort();
```

Comment:
Please give better names. We have `as` triggering `_as`, but I can't tell what either one relates to.

Using short names stops working when you have multiple objects of the same class, the names then have to reflect the role of the object.